OpenRA.Mods.Common/UtilityCommands/CheckYaml.cs

```diff
@@ -77,29 +77,8 @@ void IUtilityCommand.Run(Utility utility, string[] args)
 						}
 					}
 
-					foreach (var kv in modData.Manifest.MapFolders)
-					{
-						var name = kv.Key;
-						var classification = string.IsNullOrEmpty(kv.Value)
-							? MapClassification.Unknown : Enum<MapClassification>.Parse(kv.Value);
-
-						if (classification == MapClassification.Unknown)
-							continue;
-
-						var optional = name.StartsWith("~", StringComparison.Ordinal);
-						if (optional)
-							name = name.Substring(1);
-
-						using (var package = (IReadWritePackage)modData.ModFiles.OpenPackage(name))
-						{
-							foreach (var map in package.Contents)
-							{
-								var mapPackage = package.OpenPackage(map, modData.ModFiles);
-								if (mapPackage != null)
-									maps.Add(new Map(modData, mapPackage));
-							}
-						}
-					}
+					// Use all system maps for lint checking
+					maps = modData.MapCache.EnumerateMapsWithoutCaching().ToList();
```

Comment:
This is silently changing the command behaviour.

Please do `EnumerateMapsWithoutCaching(MapClassification.System | MapClassification.User)` to keep the current behaviour for prep.

Reply:
This can be changed to enumerate only system dirs in a followup commit that is not cherry-picked.